src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Binders/PlaceholderLocalBinder.cs

```diff
@@ -109,7 +109,9 @@ private PlaceholderLocalSymbol LookupPlaceholder(string name)
                 return null;
             }
 
-            return CreatePlaceholderLocal(_typeNameDecoder, _containingMethod, new Alias(kind, id, id, typeName));
+            // The old API (GetObjectTypeNameById) doesn't return custom type info,
+            // but the new one (GetAliases) will.
+            return CreatePlaceholderLocal(_typeNameDecoder, _containingMethod, new Alias(kind, id, id, typeName, default(CustomTypeInfo)));
```

Comment:
I added it for PR purposes.  It will be deleted when the corresponding code is deleted (i.e. shortly).